src/boot/libc64/__osMalloc.c

```diff
@@ -275,16 +416,81 @@ void __osFree(Arena* arena, void* ptr) {
 
     ArenaImpl_Lock(arena);
 
+#if MM_VERSION <= N64_JP_1_1
+    // __osFree:Unauthorized release(%08x)\n
+    (void)"__osFree:不正解放(%08x)\n";
+    // __osFree:Double release(%08x)\n
+    (void)"__osFree:二重解放(%08x)\n";
+    // __osFree:arena(%08x) and __osMallocのarena(%08x) do not match\n
+    (void)"__osFree:arena(%08x)が__osMallocのarena(%08x)と一致しない\n";
+#endif
+
+    node = (ArenaNode*)((uintptr_t)ptr - sizeof(ArenaNode));
+
+    if ((ptr != NULL) && (node->magic == NODE_MAGIC) && !node->isFree && CHECK_CORRECT_ARENA(node, arena)) {
```

Comment:
Looking at OoT's version, these should actually be split up with the strings being associated with the different conditions which I think we should do here as well.